### PR TITLE
Added Data Connectors Check Requirements to security insights

### PR DESCRIPTION
**Pull request by orgadhadas**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by t-haorga** [comment]
Hi @fengzhou-msft , I accidentally opened this PR form my personal Github account. I am a MS employee working in Azure Sentinel group. Thanks!

**Comment by fengzhou-msft** [comment]
@t-haorga please use your personal account which submitted this PR to sign CLA in the above comment.

**Comment by t-haorga** [comment]
@fengzhou-msft I don't understand what you mean, can you please elaborate?
(I never ran this command)

**Comment by t-haorga** [comment]
@fengzhou-msft seems like I don't have permissions, or I didn't understand what you meant. Can you please elaborate?

**Comment by t-haorga** [comment]
@fengzhou-msft signed

**Comment by t-haorga** [comment]
@fengzhou-msft waiting for your approval

**Comment by orgadhadas** [comment]
I am adding some changes that we applied since I opened this PR. Please do not automatically yet but do review the PR. Thanks!

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json
Comment (sanjaiganesh) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: >DataConnectorRequirements_list [](start = 24, length = 30)

update

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json
Comment (sanjaiganesh) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: >AADCheckRequirements [](start = 5, length = 20)

not used anywhere.. remove ?

Comment (sanjaiganesh) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: Is this used in some other file ?.. same with other definitions added with this change

---
In reply to: [363532156](https://github.com/Azure/azure-rest-api-specs/pull/8020#discussion_r363532156) [](ancestors = 363532156)

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json
Comment (sanjaiganesh) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: I suppose this is already being enforced at server level and even discriminator pattern may not work. hope that is the case.
otherwise, adding new required property is a breaking change.

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json
Comment (sanjaiganesh) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: Looks like this need to use discriminator pattern as others, based on kind...  for ex: this one has additional property (tenantId), compared to others that inherited from DataConnectorsCheckRequirements

---
In reply to: [363532368](https://github.com/Azure/azure-rest-api-specs/pull/8020#discussion_r363532368) [](ancestors = 363532368,363532156)

Comment (sanjaiganesh) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: >AADCheckRequirements [](start = 5, length = 20)

you may want to use discriminator pattern. i suppose we want the clients to use any of these objects (that inherit from DataConnectorsCheckRequirements) with the newly added POST.
This object has additional property of DataConnectorTenantId

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json
Comment (t-haorga) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: Can you please explain why?

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json
Comment (t-haorga) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: I added these because a test wouldn't pass, there was an error saying a discriminator must be required.
It is enforced in the server level as well.
What do you mean by "even discriminator pattern may not work" ?

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json
Comment (t-haorga) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: It is not the only thing that inherits from DataConnectorsCheckRequirements that has DataConnectorTenantId.
I added discriminators to anything that inherits from DataConnectorsCheckRequirements. Thanks

**Comment by t-haorga** [comment]
@sanjaiganesh waiting for your response

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json
Comment (sanjaiganesh) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: it is a post operation.. _List is usually for collection..   probably rename the action name to '/listDataConnectorRequirements' ?

---
In reply to: [364130217](https://github.com/Azure/azure-rest-api-specs/pull/8020#discussion_r364130217) [](ancestors = 364130217)

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json
Comment (sanjaiganesh) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: >		"SecurityRead [](start = 0, length = 15)

change tabs -to- spaces

**Comment by t-haorga** [comment]
@fengzhou-msft Please merge

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json
Comment (fengzhou-msft) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: SDK automation tests failed because the values in `x-ms-enum` do not match with `enum` values above.

Comment (t-haorga) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: Ok I fixed that now

**Comment by t-haorga** [comment]
@fengzhou-msft I think it should be fine now, but I couldn't run the tests myself

**Comment by t-haorga** [comment]
@fengzhou-msft I think this check might be stuck - it is still waiting for status report after a few hours.

**Comment by t-haorga** [comment]
@fengzhou-msft all tests passed, please merge. Thanks!

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**
